OpenRA.Mods.Common/Traits/Buildings/BuildingInfluence.cs

```diff
@@ -29,28 +30,20 @@ public BuildingInfluence(World world)
 			map = world.Map;
 
 			influence = new CellLayer<Actor>(map);
+		}
 
-			world.ActorAdded += a =>
-			{
-				var b = a.Info.TraitInfoOrDefault<BuildingInfo>();
-				if (b == null)
-					return;
-
-				foreach (var u in b.Tiles(a.Location))
-					if (influence.Contains(u) && influence[u] == null)
-						influence[u] = a;
-			};
-
-			world.ActorRemoved += a =>
-			{
-				var b = a.Info.TraitInfoOrDefault<BuildingInfo>();
-				if (b == null)
-					return;
-
-				foreach (var u in b.Tiles(a.Location))
-					if (influence.Contains(u) && influence[u] == a)
-						influence[u] = null;
-			};
+		public void AddInfluence(Actor a, IEnumerable<CPos> tiles)
```

Comment:
It can be `internal` - I assume it is supposed to be called only from Building after adding/removing from World.